 <div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom">
    <a href="https://www.mathjax.org">
    <img style="height:1.5em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" >
</a></span></div>
<div style="display:inline-block; position:absolute; top:0; left:0; color: #230e3e; font:italic 1em serif;"><a href="https://www.julialang.org">
    <img style="height:2.25em; vertical-align:top" title="Powered by Julia"
    src="./figures/julia-logo-mask.png"
    border="0" alt="Powered by Julia" >
</a></div>
</a></div>$\require{cancel} \require{mhchem} \require{bbox} \require{color} \require{enclose} \require{AMSmath} \require{AMSsymbols}$ 



In [1]:
## Jupyter notebook housekeeping
HTML("<link rel='stylesheet' href='./css/youcan.css' type='text/css' charset='utf-8' />")

HTML{String}("<link rel='stylesheet' href='./css/youcan.css' type='text/css' charset='utf-8' />")

In [2]:
using Unitful     # units package for julia

![cerulean1](./figures/2019,test2,head.png)

![cerulean1](./figures/2019,test2,c1.png)

<hr class="little"/>

On a Pv diagram, work is the area under the curve.  
1. When volume increases, the system is doing work; decreasing volume means work is being done to the system.  

2. In a cycle (like our case), the area under both curves cancels, so we only need to calculate the area bounded by the cycle. 

In [3]:
W_net = (200 - 100)u"kPa" * (6 - 1)u"m^3"
      + (400 - 200)u"kPa" * (3 - 1)u"m^3"
      + (400 - 200)u"kPa" * 0.5*(6 - 3)u"m^3" |> u"kJ"

300.0 kJ

We have the magnitude, let's get the direction. By inspection, we see that more work is going into the system (1-2-3-4-5) than out (5-1), so the net work goes __into the system__.

$$ 
\bbox[#dff0d8, 0.5em,border:2px solid #4cae4c]{
\color{#3c763d}
\small W_{net} = 300\, \ce{kJ}, \text{into the system.}}
$$

<hr class="big"/>

![cerulean1](./figures/2019,test2,c2.png)

<hr class="little"/>

With a few judicious assumptions and system selection, we can make this a classic mechanical energy problem. Just don't overthink it. My walkthrough is very much overexplained, in the interest of pointing out the underlying fundamental physics.

As usual, start with a hideous sketch.

![cerulean1](./figures/simpledam.png)

**Specify the system.**
1. What should we include in the system? Our energy flow is  

  $$\text{Water ⟶ Dam ⟶ Electricity}.$$    

  The most straightforward approach is to include the water only, then solve for $n_\ce{dam}.$
  
  $$\bbox[5px,border:2px dashed #00A]{\text{Water}} \text{ ⟶ Dam ⟶ Electricity}$$  
    
    
2. Make system boundaries coincident with the water surfaces.   

  1. Unless the height difference is extreme, we can assume both surfaces are at the same pressure.  
    
  2. We can assume the water has zero velocity. Sometimes, this is an uncomfortable assumption. It depends on specific situations. Generally, it's safe to assume zero velocity in and out OR assume the same velocity in and out. Both options are mathematically identical. 
    
    
3. Assume the water flows in and out at the same temperature. We already established equal pressure in and out. We are not at saturation conditions, so T and P are independent; thus, we can now say the water enters and leaves at the same thermodynamic state. 

  Therefore, the only energetic difference (in vs out) is mechanical energy, aka PE + KE. 

**Summarizing our assumptions,**  

<ol class = "a-prefix-list">
  <li>(a<sub></sub>) the dam is a steady state steady flow device.</li>
  <li>(a<sub></sub>) $\vec V_{in} = \vec V_{out}$</li>
  <li>(a<sub></sub>) $\left. 
      \begin{array}{l}
        P_{in} = P_{out} \\
        T_{in} = T_{out}
      \end{array} 
      \right\} \small \ce{State}_{in} = \ce{State}_{out}
      $. 
      
  <li>(a<sub></sub>) $\rho_{water} = 1000\, \ce{kg/m^3}$   </li>

</ol>

**Attack**

Mass: We have one entrance and one exit, so mass conservation is trivial.  

Energy: We apply the first law as usual. Again, I'm being overly explicit in the walkthrough.

\begin{array}{cccl}
 &\dot E_{in}& &-& & \dot E_{out} & &= \frac{d}{dt} E_{sys} \\
 \big(\dot E_{thermo} &+& \dot E_{mech} \big)_{in} &-& \big(\dot E_{thermo} + \dot E_{mech} \big)_{out} &-& \dot W_{water,out} &= \cancelto{0 \ (a_1)}{\frac{d}{dt} E_{sys}} \\
 \big( \dot E_{thermo} &+&  \ce{\dot{KE} + \dot{PE}}\big)_{in} &-& \big(\dot E_{thermo} + \ce{KE + PE}\big)_{out} &-& \dot W_{water,out} &= 0 \\[6pt]
 ~
\end{array}

Gathering like terms,  

$$
\cancelto{0 \ (a_3)}{(\dot E_{thermo,in} - \dot E_{thermo,out})} +  \cancelto{0 \ (a_2)}{\left(\, \ce{KE}_{in} - \ce{KE}_{out} \right)} + \left( \dot m_{water}gz_{in} - \dot m_{water}gz_{out}\right)  - \dot W_{water,out} = 0 \\[12pt]
{~}
$$

Cleaning up, 

$$ \dot W_{water,out} = \dot m_{water}g \left(z_{in} - z_{out}\right),  $$   

$$ \dot m_{water} = \rho_{water}\dot V_{water},  $$   

and   

$$ \eta_\ce{dam} =  \small{\frac{ \text{desired output}}{\text{required input}}} = \frac{\dot W_{electricity,out}}{\dot W_{water,out}}  .$$   



<div class="btn-info">&nbsp;<div class="badge"></div><i class="fa fa-exclamation-circle"></i><strong> Note</strong></div>
<div class="panel-info"><div class="panel-heading">   
    
For what it's worth, in this case $\small \dot E_{thermodynamic} = \dot E_{internal} + \dot E_{flow} = \dot m (e_{internal} + e_{flow}) = \dot m (u +Pv) = \dot mh$.</div></div>

In [4]:
g = 1u"gn"   # unitful.jl built-in gravitational constant
V̇_water = 1560u"m^3/s"
ρ_water = 1000u"kg/m^3"   # (a3)
Δz_dam = 216u"m"

ṁ_water = ρ_water * V̇_water

Ẇ_water_out = ṁ_water * g * Δz_dam |> u"GW"

3.3044487840000003 GW

In [5]:
Ẇ_electricity_out = 1.21u"GW"

η_dam = Ẇ_electricity_out / Ẇ_water_out

0.36617302282267716

$$ 
\bbox[#dff0d8, 0.5em,border:2px solid #4cae4c]{
\color{#3c763d}
\eta_{dam} = 36.6\%}
$$

<hr class="big"/>

![cerulean1](./figures/2019,test2,c4.png)

<hr class="little"/>

A little thought before diving in is very helpful. We need to find how much gasoline is consumed. We know that the engine receives fuel and outputs work. 

We have efficiency

\begin{align*}
 \eta = \scriptsize \frac{\text{desired output}}{\text{required input}},
\end{align*}

but it relates energy (or power). No problem:

\begin{align*}
 \eta = \frac{E_\text{desired}}{E_\text{required}} 
      = \frac{E_\text{output}}{E_\text{fuel}}
      = \frac{\dot E_\text{output}t_\text{output}}{m_\text{fuel}e_\text{fuel}},
\end{align*}

so

\begin{align*}
 m_\text{fuel} = \frac{\dot E_\text{output}t_\text{output}}{\eta e_\text{fuel}}.
\end{align*}

<div class="btn-warning">&nbsp;<div class="badge"></div><i class="fa fa-exclamation-circle"></i><strong> Reality Check</strong> </div>
<div class="panel-warning"><div class="panel-heading">
    
A quick and dirty reality check: according to our last equation, as efficiency ( $\eta$ ) goes up, fuel consumption ( $m_\text{fuel}$ ) goes down. That's the expected behavior. It's not a thorough algebra check, but it can give a little reassurance. 

Just never forget  $$\small \textit{energy = power} \times \textit{time}.$$ 
</div></div>

In [6]:
Ė = 25u"kW"
t = 10u"minute"
η = 0.28
e_fuel = 43u"MJ/kg"

m_fuel = (Ė * t) / (η * e_fuel) |> u"kg"

1.245847176079734 kg

$$ 
\bbox[#dff0d8, 0.5em,border:2px solid #4cae4c]{
\color{#3c763d}
m_{fuel} = 1.246\,\ce{kg} }
$$

<hr class="big"/>

![cerulean1](./figures/2019,test2,c5.png)

<hr class="little"/>

This is purely boundary work. Let's use 2 and 1 to represent final and inital states, respectively. 

\begin{align*}
W_{b,out}=\int^2_1 P\ dV = P \left(V_2 - V_1 \right) = mP \left(v_2 - v_1 \right)
\end{align*}

The shape of the volume matters not, only the change in volume and its relationship to pressure.

In [7]:
P = 350u"kPa"
v₁ = 0.3749u"m^3/kg"
v₂ = 0.0938u"m^3/kg"
m = 18u"kg"

Wb_out = m * P * (v₂ - v₁) |> u"kJ"

-1770.93 kJ


$$ 
\bbox[#dff0d8, 0.5em,border:2px solid #4cae4c]{
\color{#3c763d}
W_b = 1770.93\, \ce{kJ}\text{, into the system.} }
$$

<div class="btn-warning">&nbsp;<div class="badge"></div><i class="fa fa-exclamation-circle"></i><strong> Reality Check</strong> </div>
<div class="panel-warning"><div class="panel-heading">
    
Based on the information given, the volume decreased. A decrease in volume means the boundary work goes into the system.  
</div></div>

<hr class="big"/>

![cerulean1](./figures/2019,test2,c6.png)

Movement of energy driven by a temperature difference.

<hr class="big"/>

![cerulean1](./figures/2019,test2,c7.png)

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>